app/models/user_field.rb

```diff
@@ -19,7 +20,8 @@ class UserField < ActiveRecord::Base
   scope :public_fields, -> { where(show_on_profile: true).or(where(show_on_user_card: true)) }
 
   enum :requirement, { optional: 0, for_all_users: 1, on_signup: 2 }.freeze
-  enum :field_type, { text: 0, confirm: 1, dropdown: 2, multiselect: 3 }.freeze
+  enum :field_type_enum, { text: 0, confirm: 1, dropdown: 2, multiselect: 3 }.freeze
+  alias_attribute :field_type, :field_type_enum
```

Comment:
Ah wow I didn't think of this! Nice

Reply:
hmmm why are we doing the alias attribute here? is this to avoid needing to change internal code? I do worry a bit about this being a snowflake.

Reply:
Yes this is why. We do do this in another place in core, and a couple of places in chat:

https://github.com/discourse/discourse/blob/48c58499f1399baf88b43762d39bb0849eb00cf0/app/models/reviewable_history.rb#L9-L9

I think it's fine to get us over the line. We could always do the rest of the actual code change in another PR too

Reply:
sure as long as this works properly I am ok.

Reply:
I think it's safer than praying we caught all call sites, which could be hard-to-search ones like hashes passed to `#update`, etc.